dietpi/dietpi-software

```diff
@@ -8568,6 +8585,222 @@ _EOF_
+
+				cat << _EOF_ > /etc/tor/torrc
+BridgeRelay 1
+
+ORPort $orport
+
+ServerTransportPlugin obfs4 exec $(command -v obfs4proxy)
+
+ServerTransportListenAddr obfs4 0.0.0.0:$obfs4port
+
+ExtORPort auto
+_EOF_
+
+			G_CONFIG_INJECT 'NoNewPrivileges=' 'NoNewPrivileges=no' /lib/systemd/system/tor@default.service
+			G_CONFIG_INJECT 'NoNewPrivileges=' 'NoNewPrivileges=no' /lib/systemd/system/tor@.service
+			G_EXEC systemctl daemon-reload
+
+			fi
+		
+		elif [[ $G_WHIP_RETURNED_VALUE == 'Guard/Middle' ]]; then
+		
+			cat << _EOF_ > /etc/tor/torrc
+ORPort $orport
+ExitRelay 0
+SocksPort 0
+ControlSocket 0
+_EOF_
+		elif [[ $G_WHIP_RETURNED_VALUE == 'Exit' ]]; then
+			G_WHIP_BUTTON_OK_TEXT='YES'
+			G_WHIP_BUTTON_CANCEL_TEXT='NO'
+			if G_WHIP_YESNO 'WARNING: DO NOT RUN AN EXIT RELAY FROM HOME.\n
+Also, it is highly recommended that you setup exit relays on dedicated servers.\n
+Before running an exit relay, check with your hosting provider to make sure they allow it.\n
+Also, set a reverse DNS (PTR) record (tor-exit is a good name), and preferably edit your WHOIS record to show that this is a TOR exit node.\n
+Would you still like to continue?'; then
+
+			cat << _EOF_ > /etc/tor/torrc
+ORPort $orport
+ExitRelay 1
+SocksPort 0
+ControlSocket 0
+
+ExitPolicy accept *:20-21
+ExitPolicy accept *:43
+ExitPolicy accept *:53
+ExitPolicy accept *:79
+ExitPolicy accept *:80-81
+ExitPolicy accept *:88
+ExitPolicy accept *:110
+ExitPolicy accept *:143
+ExitPolicy accept *:220
+ExitPolicy accept *:389
+ExitPolicy accept *:443
+ExitPolicy accept *:464
+ExitPolicy accept *:531
+ExitPolicy accept *:543-544
+ExitPolicy accept *:554
+ExitPolicy accept *:636
+ExitPolicy accept *:706
+ExitPolicy accept *:749
+ExitPolicy accept *:873
+ExitPolicy accept *:902-904
+ExitPolicy accept *:981
+ExitPolicy accept *:989-990
+ExitPolicy accept *:991
+ExitPolicy accept *:992
+ExitPolicy accept *:993
+ExitPolicy accept *:995
+ExitPolicy accept *:1194
+ExitPolicy accept *:1220
+ExitPolicy accept *:1293
+ExitPolicy accept *:1500
+ExitPolicy accept *:1533
+ExitPolicy accept *:1677
+ExitPolicy accept *:1723
+ExitPolicy accept *:1755
+ExitPolicy accept *:1863
+ExitPolicy accept *:2082
+ExitPolicy accept *:2083
+ExitPolicy accept *:2086-2087
+ExitPolicy accept *:2095-2096
+ExitPolicy accept *:2102-2104
+ExitPolicy accept *:3690
+ExitPolicy accept *:4321
+ExitPolicy accept *:4643
+ExitPolicy accept *:5050
+ExitPolicy accept *:5190
+ExitPolicy accept *:5222-5223
+ExitPolicy accept *:5228
+ExitPolicy accept *:8008
+ExitPolicy accept *:8074
+ExitPolicy accept *:8082
+ExitPolicy accept *:8087-8088
+ExitPolicy accept *:8232-8233
+ExitPolicy accept *:8332-8333
+ExitPolicy accept *:8443
+ExitPolicy accept *:8888
+ExitPolicy accept *:9418
+ExitPolicy accept *:10000
+ExitPolicy accept *:11371
+ExitPolicy accept *:19294
+ExitPolicy accept *:19638
+ExitPolicy accept *:50002
+ExitPolicy accept *:64738
+ExitPolicy reject *:*
+_EOF_
+
+			else
+				${aSOFTWARE_INSTALL_STATE[$software_id]}=-1
```

Comment:
`$` is used for variable calls while here you want to assign a variable. Also, since you want to run an uninstall loop after this install loop, there needs to be another flag set:
```
aSOFTWARE_INSTALL_STATE[$software_id]=-1
UNINSTALL_REQUIRED=1
```

But actually, since the choice has been made before, I'm not sure if a separate YESNO makes sense here. Probably it is better to add the above warning/explanation directly into the `G_WHIP_MENU` choice loop? Either the menu text could contain further explanations (and the Exit relay warning) or, when Exit relay has been chosen, you add the dedicated YESNO directly into the same while loop so that when "NO" is chosen, users land in the parent G_WHIP_MENU to choose another relay type. This prevents the probably unexpected behaviour, when accidentally or intentionally choosing "NO", the whole already done install is uninstalled automatically.